components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java

```diff
@@ -140,12 +140,24 @@ public String renderPage(HttpRequest request, HttpResponse response) {
                     uriWithoutContextPath.substring(0, uriWithoutContextPath.length() - 1) :
                     (uriWithoutContextPath + "/");
             if (hasPage(correctedUriWithoutContextPath)) {
-                // Redirecting to the correct page.
-                String correctedUri = request.getContextPath() + correctedUriWithoutContextPath;
-                if (request.getQueryString() != null) {
-                    correctedUri = correctedUri + '?' + request.getQueryString();
+                if (request.getMethod().equals("GET")) {
+                    // Redirecting to the correct page.
+                    String correctedUri = request.getContextPath() + correctedUriWithoutContextPath;
+                    if (request.getQueryString() != null) {
+                        correctedUri = correctedUri + '?' + request.getQueryString();
+                    }
+                    throw new PageRedirectException(correctedUri, e);
+                } else {
+                    // If GET, we correct, since this can be an end-user error. But if POST it's the responsibility of
+                    // the dev to use correct URL. Because HTTP POST is not well supported. See :
+                    // https://softwareengineering.stackexchange.com/questions/99894/why-doesnt-http-have-post-redirect
```

Comment:
Lets change this line to `See https://softwareengineering.stackexchange.com/q/99894`, shorter URL